packages/devtools/lib/src/framework/framework.dart

```diff
@@ -39,15 +45,93 @@ class Framework {
 
   Screen current;
 
+  Screen previous;
+
   StatusLine globalStatus;
   StatusLine pageStatus;
   StatusLine auxiliaryStatus;
   ActionsContainer globalActions;
   ConnectDialog connectDialog;
+  ImportMessage importMessage;
   AnalyticsOptInDialog analyticsDialog;
 
   final Map<Screen, CoreElement> _screenContents = {};
 
+  void _initDragDrop() {
+    window.addEventListener('dragenter', (e) => _onDragEnter(e), false);
+    window.addEventListener('dragover', (e) => _onDragOver(e), false);
+    window.addEventListener('dragleave', (e) => _onDragLeave(e), false);
+    window.addEventListener('drop', (e) => _onDrop(e), false);
+  }
+
+  void _onDragEnter(MouseEvent event) {
+    final Element dropTarget = event.target;
+    dropTarget.classes.add('over');
+  }
+
+  void _onDragOver(MouseEvent event) {
+    // This is necessary to allow us to drop.
+    event.preventDefault();
+    event.dataTransfer.dropEffect = 'move';
+  }
+
+  void _onDragLeave(MouseEvent event) {
+    final Element dropTarget = event.target;
+    dropTarget.classes.remove('over');
+  }
+
+  void _onDrop(MouseEvent event) async {
+    // Stop the browser from redirecting.
+    event.preventDefault();
+
+    final List<File> files = event.dataTransfer.files;
+    if (files.length > 1) {
+      toast('You cannot import more than one file.');
+      return;
+    }
+
+    final droppedFile = files.first;
+    if (droppedFile.type != 'application/json') {
+      toast('${droppedFile.type} is not a supported file type.\n Please import '
+          'a .json file.');
+      return;
+    }
+
+    final FileReader reader = FileReader();
+    reader.onLoad.listen((_) {
+      final Map<String, dynamic> import = jsonDecode(reader.result);
```

Comment:
We also need to handle if the file being dragged isn't json. We should display a user friendly message for that case to avoid confusion.

Reply:
a file might have type "application/json" but not parse.
To test, you could manually edit your json file to be invalid and see what error you get.

Reply:
will address this case on a follow up CL that will also move all the drag-and-drop import code to it's own file import_manager.dart. This will make it easier to add unit tests.